src/components/landing/pages/history/history.tsx

```diff
@@ -102,22 +102,79 @@ export const History: React.FC = () => {
     historyWrite([])
   }, [historyWrite, user])
 
-  const syncClick = useCallback((entryId: string): void => {
-    console.log(entryId)
+  const uploadAll = useCallback((): void => {
+    const newHistory = mergeEntryArrays(localHistoryEntries, remoteHistoryEntries)
+    if (user) {
+      setHistory(newHistory)
+        .then(() => {
+          setRemoteHistoryEntries(newHistory)
+          setLocalHistoryEntries([])
+          historyWrite([])
+        })
+        .catch(() => setError('setHistory'))
+    }
+  }, [historyWrite, localHistoryEntries, remoteHistoryEntries, user])
+
+  const deleteClick = useCallback((entryId: string, location: HistoryEntryOrigin): void => {
+    if (user) {
+      deleteNote(entryId)
+        .then(() => {
+          if (location === HistoryEntryOrigin.LOCAL) {
+            setLocalHistoryEntries(entries => entries.filter(entry => entry.id !== entryId))
+          } else if (location === HistoryEntryOrigin.REMOTE) {
+            setRemoteHistoryEntries(entries => entries.filter(entry => entry.id !== entryId))
+            setHistory(remoteHistoryEntries)
+              .catch(() => setError('setHistory'))
```

Comment:
I don't know if this is necessary. I would expect, that the api handles the removal from the histores itself.